### ISSUE TITLE
pyclient unable to get serial ID from server running on AWS

### ISSUE THREAD

**Issue by hppritcha** [open-mpi/mtt#562]
The pyclient seems to be unable to get a serial ID from the AWS server.  I've tried this using a variety of setups, both behind and not behind a firewall.  Using the --debug flag, this output appears when the trying to get a serial id:

```
Reporter: IU database
<<<<<<<---------------- Payload (Start) -------------------------->>>>>>
{
    "serial": "serial"
}
<<<<<<<---------------- Payload (End  ) -------------------------->>>>>>
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): mtt.open-mpi.org
/opt/hi-master/Python-3.6.2/lib/python3.6/site-packages/urllib3/connectionpool.py:858: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
DEBUG:urllib3.connectionpool:https://mtt.open-mpi.org:443 "POST /serial HTTP/1.1" 404 287
<<<<<<<---------------- Response -------------------------->>>>>>
Result: 404: text/html; charset=iso-8859-1
{'Date': 'Thu, 31 Aug 2017 15:50:13 GMT', 'Server': 'Apache/2.4.18 (Ubuntu)', 'Content-Length': '287', 'Keep-Alive': 'timeout=5, max=100', 'Connection': 'Keep-Alive', 'Content-Type': 'text/html; charset=iso-8859-1'}
Not Found
<<<<<<<---------------- Raw Output (Start) ---------------->>>>>>
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>404 Not Found</title>
</head><body>
<h1>Not Found</h1>
<p>The requested URL /serial was not found on this server.</p>
<hr>
<address>Apache/2.4.18 (Ubuntu) Server at mtt.open-mpi.org Port 443</address>
</body></html>

<<<<<<<---------------- Raw Output (End  ) ---------------->>>>>>
Error: Unable to get a client serial (rtn=-1)
<<<<<<<<<<<<<<<<<<<<<<<<<>>>>>>>>>>>>>>>>>>>>>>>>>>
```

The IU report section my ini file looks like this:
```
plugin = IUDatabase
#module = IUDatabase

realm = OMPI
username = lanl
password = XXXX
mtt_platform = hickok
url = https://mtt.open-mpi.org
#url = https://mtt.open-mpi.org/submit/cpy
mttdatabase_debug_filename = mttdb_debug_file
mttdatabase_debug_server = 1
```
I also tried using the submit/cpy url option and that resulted in this debug output:
```
Reporter: IU database
<<<<<<<---------------- Payload (Start) -------------------------->>>>>>
{
    "serial": "serial"
}
<<<<<<<---------------- Payload (End  ) -------------------------->>>>>>
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): mtt.open-mpi.org
/opt/hi-master/Python-3.6.2/lib/python3.6/site-packages/urllib3/connectionpool.py:858: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
DEBUG:urllib3.connectionpool:https://mtt.open-mpi.org:443 "POST /submit/cpy/serial HTTP/1.1" 404 298
<<<<<<<---------------- Response -------------------------->>>>>>
Result: 404: text/html; charset=iso-8859-1
{'Date': 'Thu, 31 Aug 2017 15:54:30 GMT', 'Server': 'Apache/2.4.18 (Ubuntu)', 'Content-Length': '298', 'Keep-Alive': 'timeout=5, max=100', 'Connection': 'Keep-Alive', 'Content-Type': 'text/html; charset=iso-8859-1'}
Not Found
<<<<<<<---------------- Raw Output (Start) ---------------->>>>>>
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>404 Not Found</title>
</head><body>
<h1>Not Found</h1>
<p>The requested URL /submit/cpy/serial was not found on this server.</p>
<hr>
<address>Apache/2.4.18 (Ubuntu) Server at mtt.open-mpi.org Port 443</address>
</body></html>

<<<<<<<---------------- Raw Output (End  ) ---------------->>>>>>
Error: Unable to get a client serial (rtn=-1)
<<<<<<<<<<<<<<<<<<<<<<<<<>>>>>>>>>>>>>>>>>>>>>>>>>>
```
I then tried using 
```
url = https://mtt.open-mpi.org/submit/cpy/api
```
and got this debug output:
```
Reporter: IU database
<<<<<<<---------------- Payload (Start) -------------------------->>>>>>
{
    "serial": "serial"
}
<<<<<<<---------------- Payload (End  ) -------------------------->>>>>>
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): mtt.open-mpi.org
/opt/hi-master/Python-3.6.2/lib/python3.6/site-packages/urllib3/connectionpool.py:858: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
DEBUG:urllib3.connectionpool:https://mtt.open-mpi.org:443 "POST /submit/cpy/api//serial HTTP/1.1" 400 50
<<<<<<<---------------- Response -------------------------->>>>>>
Result: 400: text/plain
{'Date': 'Thu, 31 Aug 2017 15:55:48 GMT', 'Server': 'Apache/2.4.18 (Ubuntu)', 'Content-Length': '50', 'Content-Type': 'text/plain', 'Connection': 'close'}
Bad Request
<<<<<<<---------------- Raw Output (Start) ---------------->>>>>>
Absolute URI not allowed if server is not a proxy.
<<<<<<<---------------- Raw Output (End  ) ---------------->>>>>>
Error: Unable to get a client serial (rtn=-1)
```

**Comment by jjhursey** [comment]
This should be correct: `url = https://mtt.open-mpi.org/submit/cpy/api`

The error `Absolute URI not allowed if server is not a proxy.` is probably the source of the problem on the AWS server.

**Comment by hppritcha** [comment]
@jjhursey have you had a chance to look at fixing the server problem at AWS?

**Comment by jjhursey** [comment]
I have not had a chance yet sorry. :/

**Comment by jjhursey** [comment]
Note for myself:
CherryPy 8.9.0 works, but CherryPy 9.0.0 and later do not. The check for proxy was added, but the switch to tell CherryPy it is in proxy mode does not seem to work.
I reverted the server to v8.9.0 and now the error message:
`Absolute URI not allowed if server is not a proxy.`
Is no longer displayed. I need to keep digging to see how to get around that...

Now the server will return a serial, but there is a Python Client side problem that we hit right after that. I'll looking into it.
